# Generated by Django 5.0.7 on 2024-07-24 12:16

import django.db.models.deletion
from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('auth_app', '0003_audit_status'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.CreateModel(
            name='AuditForm',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('inp', models.CharField(max_length=16)),
                ('audit', models.CharField(max_length=100)),
                ('case_error', models.CharField(blank=True, max_length=100, null=True)),
                ('case_status', models.CharField(max_length=100)),
                ('status', models.CharField(default='draft', max_length=10)),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='ComponentWithError',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('inp', models.CharField(max_length=16)),
                ('audit', models.CharField(max_length=100)),
                ('case_error', models.CharField(max_length=100)),
                ('case_status', models.CharField(max_length=100)),
                ('remarks', models.TextField(blank=True, null=True)),
                ('remarks_for_operations', models.TextField(blank=True, null=True)),
                ('audit_form', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='components_with_error', to='auth_app.auditform')),
            ],
        ),
        migrations.CreateModel(
            name='ComponentWithoutError',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('inp', models.CharField(max_length=16)),
                ('audit', models.CharField(max_length=100)),
                ('case_status', models.CharField(max_length=100)),
                ('remarks', models.TextField(blank=True, null=True)),
                ('remarks_for_operations', models.TextField(blank=True, null=True)),
                ('audit_form', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='components_without_error', to='auth_app.auditform')),
            ],
        ),
        migrations.CreateModel(
            name='ErrorCase',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('audit', models.CharField(max_length=100)),
                ('error_definition', models.CharField(max_length=100)),
                ('final_error_category', models.CharField(max_length=100)),
                ('remarks', models.TextField(blank=True, null=True)),
                ('remarks_for_operations', models.TextField(blank=True, null=True)),
                ('audit_form', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='error_cases', to='auth_app.auditform')),
            ],
        ),
        migrations.DeleteModel(
            name='Audit',
        ),
    ]
